crates/polars-stream/src/nodes/zip.rs

```diff
@@ -288,7 +301,8 @@ impl ComputeNode for ZipNode {
                 for input_head in &mut self.input_heads {
                     out.push(input_head.take(common_size));
                 }
-                let out_df = concat_df_horizontal(&out, false)?;
+                let strict_concat = matches!(self.extend_behavior, ExtendBehavior::Raise);
```

Comment:
This match is unnecessary, you can always enable strict concat here. Our inputs at this point in the code should always have the same length.